tools/reflect/src/SvLocalParam.cpp

```diff
@@ -0,0 +1,77 @@
+// SPDX-FileCopyrightText: Michael Popoloski
+// SPDX-License-Identifier: MIT
+
+#include "SvLocalParam.h"
+
+void unwrapUnpackedArray(const std::span<const slang::ConstantValue> constantValues,
+                         std::vector<std::vector<uint64_t>>& values, uint64_t& biggestElementSize) {
+    if (constantValues.front().isUnpacked())
+        for (const auto& unpackedArray : constantValues)
+            unwrapUnpackedArray(unpackedArray.elements(), values, biggestElementSize);
+    else if (constantValues.front().isInteger()) {
+        std::vector<uint64_t> collectedValues;
+        for (const auto& value : constantValues) {
+            if (!value.isInteger())
+                SLANG_THROW(std::runtime_error(
+                    "Found a non integer member while reflecting this parameter"));
+            if (value.getBitstreamWidth() > 64)
+                SLANG_THROW(std::runtime_error(
+                    "Found a value bigger than 64 bits while reflecting this parameter"));
+            biggestElementSize = std::max(biggestElementSize, value.getBitstreamWidth());
+            collectedValues.emplace_back(*value.integer().getRawPtr());
+        }
+        values.emplace_back(collectedValues);
+    }
+}
+
+void SvLocalParam::toCpp(HppFile& hppFile, std::string_view, const SvAliases&, bool) const {
+    std::string parameterName = isCppReserved(parameter.name) ? fmt::format("_{}", parameter.name)
+                                                              : std::string(parameter.name);
+    if (parameter.getValue().isUnpacked()) {
+        std::vector<std::vector<uint64_t>> unpackedArrays;
+        uint64_t biggestSize = 0;
+        SLANG_TRY {
+            unwrapUnpackedArray(parameter.getValue().elements(), unpackedArrays, biggestSize);
+        }
+        SLANG_CATCH(const std::runtime_error& error) {
```

Comment:
The exception object gets removed when exceptions are disabled, so you then can't refer to it in the format() message below. You would need to put that call between a `#if __cpp_exceptions` so that the no-exception build will pass.

Reply:
Right, done so in the last commit :+1: